service/header/core_exchange_test.go

```diff
@@ -13,14 +14,18 @@ import (
 )
 
 func TestCoreExchange_RequestHeaders(t *testing.T) {
-	fetcher := createCoreFetcher(t)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	t.Cleanup(cancel)
+
+	fetcher, err := createCoreFetcher(ctx, t)
+	require.NoError(t, err)
 	store := mdutils.Mock()
 
 	// generate 10 blocks
 	generateBlocks(t, fetcher)
```

Comment:
The referenced failures are on line 24, which is after generateBlocks. The generateBlocks awaits for 10 blocks which also includes waiting for the `IsRunning` to become true, so we were already indirectly waiting for mock to become running and I doubt this will fix issues on line 24, but it does not hurt to have this,

Reply:
Very weird then that this happens. It's likely a problem with the mock core node itself then?

Reply:
> It's likely a problem with the mock core node itself then?

I think so

Reply:
@renaynay, we can either debug the issue for them and contribute the fix or just embrace and live with it

Reply:
😞  I don't want to embrace and live with a flakey ci pipeline. Let's at least open an issue and then see if they address it.

Reply:
Agreed. One thing though. They might not have it at all rn btw. Their code is constantly changing and we are not updating it at all.

Reply:
> P.S. If Only Tendermint Was A Library (IOTWAL)

lol IOTWAL

We won't be able to update to the latest version of tendermint until at least after testnet, and even then we will have to wait until the cosmos-sdk uses some version of v0.36.x

I am planning on working on a better way to test the application and the rpc for it this week. We need something that doesn't require running a full tendermint node in order for the CI to not be so dang flaky. 

I'm out of the loop on this, do we have a test suite that runs multiple tests around a single tendermint node, or do we boot one up for each test?